### ISSUE TITLE
Upload: interruptUpload restarts the upload (restarts life cycle events)

### ISSUE THREAD

**Issue by sascha-frinken** [vaadin/framework#10545]
**Vaadin Framework version:** 8.2.1

Calling `interruptUpload` does not stop the upload but restarts the whole life cycle again.  

**Reproducable sample:**  
1. Go to: https://demo.vaadin.com/sampler/#ui/data-input/other/upload
2. upload a file
3. click cancel

(reproducable with local  Vaadin Framework version 8.2.1)

**Expected behaviour:**
(In the demo upload info window)
Current state changes to `Idle`

**Actual behaviour:**
Current state remains `Uploading`
Line counter starts from 0

**Comment by tsuoanttila** [comment]
Please test this issue with Vaadin `8.3.0.beta2` which is available in the `vaadin-prereleases` maven repository. The Sampler will be updated once we get the final release of 8.3 out.

```
<repository>
	<id>vaadin-prereleases</id>
	<url>http://maven.vaadin.com/vaadin-prereleases</url>
</repository>
```

**Comment by sascha-frinken** [comment]
Sorry - same behaviour with Vaadin `8.3.0.beta2`

**Comment by melanieretter** [comment]
Same behaviour also with Vaadin 7.7.13.
Are there any actions to fix this in Vaadin 7?

**Comment by tsuoanttila** [comment]
After looking into Upload component and it's behaviour in various cases, I found out that at least Jetty 9.3 and 9.4 are not working properly when you interrupt the upload. I will test this issue while I'm at it with a couple of different servlet containers to check if it is the same case here as well.

**Comment by sascha-frinken** [comment]
I see the error in the embedded Tomcat (v8.5) of a SpringBoot project (v1.5).

**Comment by lmenko** [comment]
What is the status on this issue? it is still reproducible in Vaadin 8.4.2. It is especially annoying because we need to interrupt the upload component if the user selects a file bigger than the maximal allowed size.

**Comment by lmenko** [comment]
This is related to https://github.com/vaadin/framework/issues/5165 which is closed but shouldn't be.

**Comment by sascha-frinken** [comment]
@tsuoanttila  Any update?

**Comment by TatuLund** [comment]
I did some testing. The code in Sampler is broken. Canceling upload requires that you click cancel button multiple times. I also tried to replicate issue in simple test application, where I interrupted upload in started listener if the file is too big, which on the other hand somewhat worked ok, but events were just fired multiple times.

**Comment by sahalsaad** [comment]
The demo seems fine now. This issue fixed?

**Comment by TatuLund** [comment]
This should be fixed by 8.9.2
